File: insee-promotion-client/app/components/enum/CustomerStatus.js


export class CustomerStatus{

    constructor(status) {
        this.status = status;
    }

    static findByStatus(status) {
        switch(status) {
            case 1 : return NEED_REVIEW;
            case 3 : return REJECTED;
            case 2 : return APPROVED;
            default: return null;
        }
    }

    getStatus() {
        return this.status;
    }

    getColor() {
        switch(this.status) {
            case 1 : return '#b71c1c';
            case 3 : return '#778899';
            case 2 : return '#088dcd';
            default: return null;
        }
    }

    getName() {
        switch(this.status) {
            case 1 : return 'Cần xét duyệt';
            case 3 : return 'Đã từ chối';
            case 2 : return 'Đã chấp nhận';
            default: return null;
        }
    }
}

export const NEED_REVIEW = new CustomerStatus(1);
export const REJECTED = new CustomerStatus(3);
export const APPROVED = new CustomerStatus(2);

File: ZaloBotServer/src/main/java/com/vn/insee/zalobot/common/status/CustomerStatus.java
package com.vn.insee.zalobot.common.status;

public enum CustomerStatus {
    REVIEWING(1), APPROVED(2), REJECTED(3);
    private int status;

    CustomerStatus(int status) {
        this.status = status;
    }

    public int getStatus() {
        return status;
    }

    public static CustomerStatus findByStatus(int status) {
        switch (status) {
            case 1 : return REVIEWING;
            case 2 : return APPROVED;
            case 3 : return REJECTED;
            default: return null;
        }
    }

}
